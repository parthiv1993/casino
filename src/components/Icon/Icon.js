import PropTypes from 'prop-types';
import { StyledIcon } from './Icon.styles';

const Icon = ({ iconName, addMargin }) => {
  const className = `${iconName}`;
  return <StyledIcon className={className} addMargin={addMargin} />;
};

Icon.propTypes = {
  iconName: PropTypes.string,
  addMargin: PropTypes.bool
};

export default Icon;
