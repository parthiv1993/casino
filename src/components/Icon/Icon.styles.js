import styled from 'styled-components';

export const StyledIcon = styled.i`
  ${({ addMargin }) =>
    addMargin
      ? `
        margin-inline: 4px;
    `
      : ``};
`;
