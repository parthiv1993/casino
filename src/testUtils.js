import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
