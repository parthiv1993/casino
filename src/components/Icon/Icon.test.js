import { renderComponentWithLightTheme } from '../../testUtils';
import Icon from './Icon';

describe('Icon', () => {
  it('should match snapshot', () => {
    const { container } = renderComponentWithLightTheme(<Icon iconName="fa-username" />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when margin is added', () => {
    const { container } = renderComponentWithLightTheme(<Icon iconName="fa-username" addMargin />);
    expect(container).toMatchSnapshot();
  });
});
