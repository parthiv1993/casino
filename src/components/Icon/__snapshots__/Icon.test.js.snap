// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<div>
  <i
    class="fa-username"
  />
</div>
`;

exports[`Icon should match snapshot when margin is added 1`] = `
.c0 {
  margin-inline: 4px;
}

<div>
  <i
    class="c0 fa-username"
  />
</div>
`;
